File: Programming Basics/Programming Basics - C#/Exercises/05. Simple Loops/05. Simple Loops/Half Sum Element/Half Sum Element.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Half_Sum_Element
{
    class Program
    {
        static void Main(string[] args)
        {
            var n = int.Parse(Console.ReadLine());
            var sum = 0;
            var max = 0;

            for (int i = 0; i < n; i++)
            {
                var num = int.Parse(Console.ReadLine());
                if (num > max)
                {
                    max = num;
                }
                sum = sum + num;
            }

            sum = sum - max;
            if (sum == max)
            {
                Console.WriteLine("Yes");
                Console.WriteLine("Sum = " + sum);
            }

            else
            {
                Console.WriteLine("No");
                Console.WriteLine("diff =" + Math.Abs(max - sum));
            }
        }
    }
}

File: Programming Basics/Programming Basics - JavaScript/Exercises/05. Simple Loops/10. Half Sum Element/HalfSumElement.js
function calculateHalfSumOfElement(args) {
    let n = Number(args[0]);
    let sum = 0;
    let max = 0;

    for (let i = 1; i <= n; i++) {
        let num = Number(args[i]);
        if (num > max) {
            max = num;
        }
        sum += num;
    }

    sum -= max;
    if (sum == max) {
        console.log("Yes");
        console.log("Sum = " + sum);
    }

    else {
        console.log("No");
        console.log("Diff = " + Math.abs(max - sum));
    }
}

/* Uncomment to test:
calculateHalfSumOfElement(["7", "3", "4", "1", "1", "2", "12", "1"]);
calculateHalfSumOfElement(["4", "6", "1", "2", "3"]);
calculateHalfSumOfElement(["3", "1", "1", "10"]);
calculateHalfSumOfElement(["3", "5", "5", "1"]);
calculateHalfSumOfElement(["3", "1", "1", "1"]);
    */